catalog/rocks/catalog.go

```diff
@@ -0,0 +1,338 @@
+package rocks
+
+import (
+	"context"
+	"time"
+)
+
+// Basic Types
+
+// DiffType represents a changed state for a given entry (added, removed, changed, conflict)
+type DiffType uint8
+
+const (
+	DiffTypeAdded DiffType = iota
+	DiffTypeRemoved
+	DiffTypeChanged
+	DiffTypeConflict
+)
+
+// function/methods receiving the following basic types could assume they passed validation
+type (
+	// StorageNamespace is the URI to the storage location
+	StorageNamespace string
+
+	// RepositoryID is an identifier for a repo
+	RepositoryID string
+
+	// Path represents a logical path for an entry
+	Path string
+
+	// Ref could be a commit ID, a branch name, a Tag
+	Ref string
+
+	// TagID represents a named tag pointing at a commit
+	TagID string
+
+	// CommitID is a content addressable hash representing a Commit object
+	CommitID string
+
+	// BranchID is an identifier for a branch
+	BranchID string
+
+	// TreeID represents a snapshot of the tree, referenced by a commit
+	TreeID string
+
+	// StagingToken represents a namespace for writes to apply as uncommitted
+	StagingToken string
+
+	// CommonPrefix represents a path prefixing one or more Entry objects
+	CommonPrefix string
+
+	// Metadata key/value strings to holds metadata information on entry and commit
+	Metadata map[string]string
+)
+
+// Repository represents repository metadata
+type Repository struct {
+	StorageNamespace StorageNamespace
+	CreationDate     time.Time
+	DefaultBranchID  BranchID
+}
+
+type RepositoryRecord struct {
+	RepositoryID RepositoryID
+	*Repository
+}
+
+// Entry represents metadata or a given object (modified date, physical address, etc)
+type Entry struct {
+	LastModified time.Time
```

Comment:
```suggestion
	CreationDate time.Time
```
Can we ever change an entry?